python/snewpy/models/ccsn.py

```diff
@@ -172,583 +109,291 @@ def __new__(cls, filename=None, *, progenitor_mass=None, revival_time=None, meta
-    """
-
-    param = {'progenitor_mass': (list(range(12, 34)) +
+@RegistryModel(
+    progenitor_mass = Parameter(values=(list(range(12, 34)) +
                                  list(range(35, 61, 5)) +
-                                 [70, 80, 100, 120]) * u.Msun,
-             'eos': ['HShen', 'LS220']}
-
-    _param_abbrv = {'progenitor_mass': '[12..33, 35..5..60, 70, 80, 100, 120] solMass',
-                    'eos': ['HShen', 'LS220']}
-
-    @_warn_deprecated_filename_argument
-    def __new__(cls, base=None, mass=None, eos='LS220', *, progenitor_mass=None):
-        """Model Initialization.
+                                 [70, 80, 100, 120]) * u.Msun, 
+                                desc_values='[12..33, 35..5..60, 70, 80, 100, 120] solMass'
+                               ),
+    eos = ['HShen', 'LS220']
+)
+class _OConnor_2013_new(loaders.OConnor_2013):
+    """Model based on the black hole formation simulation in `O'Connor & Ott (2013) <https://arxiv.org/abs/1207.1100>`_.
+    """    
+    def __init__(self, eos:str, progenitor_mass:u.Quantity):
+        # Load from Parameters
+        filename = f'{eos}_timeseries.tar.gz'
+        return super().__init__(filename=filename, metadata=self.metadata)
 
+class OConnor_2013(legacy_filename_initialization(_OConnor_2013_new)):
+    @deprecated('base', 'mass')
+    def __init__(self, base=None, mass=None, eos='LS220', *, progenitor_mass=None):
+        """
         Parameters
         ----------
         base : str
             Absolute or relative path folder with model data. This argument is deprecated.
         mass: int
             Mass of model progenitor in units Msun. This argument is deprecated.
-        eos: str
-            Equation of state. Valid values are {eos}.
-
-        Other Parameters
-        ----------------
-        progenitor_mass: astropy.units.Quantity
-            Mass of model progenitor in units Msun. Valid values are {progenitor_mass}.
-
-        Raises
-        ------
-        FileNotFoundError
-            If a file for the chosen model parameters cannot be found
-        ValueError
-            If a combination of parameters is invalid when loading from parameters
-
         """
-        # TODO: (For v2.0) Change `base` to filename, move compressed model files to OConnor_2013 model folder
-        if mass is not None:
-            warn(f'Argument `mass` of type int is deprecated. To initialize this model, use keyword arguments '
-                 f'{list(cls.param.keys())}. See {cls.__name__}.param, {cls.__name__}.get_param_combinations() for more info',
-                 category=DeprecationWarning, stacklevel=2)
+        if mass:
+            progenitor_mass = mass*u.Msun
+        if(base):
+            self.metadata = {'Progenitor mass':progenitor_mass,
+                             'EOS':eos}
+            
+            return super().__init__(filename=base, eos=eos, progenitor_mass=progenitor_mass)
```

Comment:
```suggestion
        if base:
            self.metadata = {'Progenitor mass': progenitor_mass,
                             'EOS': eos}
            filename = f'{base}/{eos}_timeseries.tar.gz'
            return super().__init__(filename=filename, eos=eos, progenitor_mass=progenitor_mass)
```

Unfortunately, there is still an issue with `OConnor_2013` when called as in the previously mentioned example:
```Python
OConnor_2013("/path/to/snewpy/models/OConnor_2013/", 12)  # using positional arguments for `base` and `mass`
```
Here, the `base` argument to the original `__init__` function gives the directory with model files; the class then uses the `eos` to generate the path of the `.tar.gz` file (this is now in `_OConnor_2013_new.__init__`). However, the above function call skips over `_OConnor_2013_new.__init__` and passes the base directory directly to the loader class, which then fails with an `IsADirectoryError`.

This suggestion does fix the `IsADirectoryError`, at the cost of duplicating that one line of filename logic. I think that’s preferable to making changes to any of these decorators, though, which could have subtle unintended side effects on other model classes.